indico/modules/designer/controllers.py

```diff
@@ -340,6 +356,48 @@ def _process_POST(self):
         return jsonify_data()
 
 
+class RHLinkDesignerTemplate(RHModifyDesignerTemplateBase):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.template,
+            lambda self: self.regform
+        }
+    }
+
+    def _process_args(self):
+        RHModifyDesignerTemplateBase._process_args(self)
+        regform_id = request.view_args['reg_form_id']
+        self.regform = (RegistrationForm.query
+                        .with_parent(self.template.event)
+                        .filter_by(id=regform_id, is_deleted=False)
+                        .first_or_404())
+
+    def _process(self):
+        if not can_link_to_regform(self.template, self.regform):
+            raise ValueError('Cannot link to the specified registration form.')
+
+        self.template.link_regform(self.regform)
+        self.template.event.log(EventLogRealm.event, LogKind.positive, 'Designer',
+                                'Badge template linked to registration form', session.user,
+                                data={'Template': self.template.title, 'Registration Form': self.regform.title})
+        flash(_('Template successfully linked.'), 'success')
+        return jsonify_data(html=_render_template_list(self.target, event=self.event_or_none))
+
+
+class RHUnLinkDesignerTemplate(RHModifyDesignerTemplateBase):
+    def _process(self):
+        regform = self.template.registration_form
+        if not regform:
+            raise ValueError('This template is not linked to any registration form.')
+
+        self.template.unlink_regform()
```

Comment:
this is quite destructive, we should have a confirmation prompt in the UI for unlinking. or should we even allow unlinking as long as there are custom placeholders?

Reply:
ah hm, maybe i unconsciously clicked "yes" :D

anyway, i think preventing is better, i can't see a good reason why you'd add custom fields and then unlink... "least surprise for the user" is generally good. we all know how much users read confirmation prompts (\*cough\* event deletion \*cough\*)